plutus-pab/src/Cardano/Protocol/Socket/Mock/Client.hs

```diff
@@ -0,0 +1,191 @@
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE NamedFieldPuns    #-}
+{-# LANGUAGE TypeFamilies      #-}
+
+module Cardano.Protocol.Socket.Mock.Client where
+
+import qualified Data.ByteString.Lazy                                as LBS
+import           Data.Time.Units                                     (Second, TimeUnit, toMicroseconds)
+import           Data.Void                                           (Void)
+
+import           Control.Concurrent
+import           Control.Concurrent.STM
+import           Control.Monad.Catch                                 (catchAll)
+import           Control.Tracer
+
+import           Ouroboros.Network.Block                             (Point (..))
+import qualified Ouroboros.Network.Protocol.ChainSync.Client         as ChainSync
+import qualified Ouroboros.Network.Protocol.LocalTxSubmission.Client as TxSubmission
+
+import           Ledger.TimeSlot                                     (SlotConfig, currentSlot)
+import           Ouroboros.Network.IOManager
+import           Ouroboros.Network.Mux
+import           Ouroboros.Network.NodeToClient                      (NodeToClientProtocols (..), connectTo,
+                                                                      versionedNodeToClientProtocols)
+import           Ouroboros.Network.Snocket
+import           Ouroboros.Network.Socket
+
+import           Cardano.Protocol.Socket.Type
+import           Ledger                                              (Block, Slot (..), Tx (..))
+
+data ChainSyncHandle = ChainSyncHandle
+    { cshCurrentSlot :: IO Slot
+    , cshHandler     :: Block -> Slot -> IO ()
+    }
+
+newtype TxSendHandle = TxSendHandle
+    { tshQueue :: TQueue Tx }
+
+-- | Queue a transaction to be sent to the server.
+queueTx ::
+    TxSendHandle
+ -> Tx
+ -> IO ()
+queueTx TxSendHandle { tshQueue } tx =
+    atomically (writeTQueue tshQueue tx)
+
+getCurrentSlot :: ChainSyncHandle -> IO Slot
+getCurrentSlot = cshCurrentSlot
+
+-- | Run the chain sync protocol to get access to the current slot number.
+runChainSync' :: FilePath
```

Comment:
should probably be called `runChainSyncMock'` (same for `runChainSync` below), just to make it abundantly clear that these are the emulator txn types

Reply:
I thought it was cool using the same name from different packages, but can do.

Reply:
True. The risk of someone confusing the two probably isn't that high. So I'm happy to leave it as it is.